condor_history_to_prometheus.py

```diff
@@ -120,8 +174,15 @@ def query_collectors(collectors, metrics, options):
 
     prometheus_client.start_http_server(options.port)
 
-    last_job = {'ClusterId': None, 'EnteredCurrentStatus': None}
-    
-    while True:
-        query_collectors(args, metrics, options)
-        time.sleep(options.interval)
\ No newline at end of file
+    if options.collectors:
+        last_job = defaultdict(dict)
+        
+        for collector in args:
+            schedd_ads = locate_schedds(collector)
+            for s in schedd_ads:
+                last_job[s.get('Name')] = {'ClusterId': None, 'EnteredCurrentStatus': None}
+
+        while True:
+            for collector in args:
+                query_collector(collector, metrics, last_job)
+            time.sleep(options.interval)
```

Comment:
One thing I thought of: because it takes a while to do the metrics collection, this sleep time should be the `interval` minus that time, to keep it roughly consistent.  Or you could loop much more often, and only do the query when the `interval` is hit (maybe `(time.time() - start_time) % interval < 1`?).